### PR TITLE
hungtran - w1-d4 event - listener

### PR DESCRIPTION
**Pull request by at-hungtran2**
#14

### CODE REVIEW

**Comment by at-hungtran2** [comment]
@at-vinhhuynh  please review it. thanks

File: app/src/at-hungtran/java/asiantech/internship/summer/LoginActivity.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        mEdtPwd.setTransformationMethod(PasswordTransformationMethod.getInstance());
+
+        if (!isValidUsername(username) || !isValidPassword(password) || !isValidEmail(email)) {
+            btnLogin.setVisibility(View.GONE);
+        } else {
+            btnLogin.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private boolean isValidUsername(String mUsername) {
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/java/asiantech/internship/summer/LoginActivity.java]: `mUsername `  -> `userName`

File: app/src/at-hungtran/java/asiantech/internship/summer/LoginActivity.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        mPattern = Pattern.compile(USERNAME_PATTERN);
+        mMatcher = mPattern.matcher(mUsername);
+
+        return mMatcher.matches();
+
+    }
+
+    private boolean isValidPassword(String mPassword) {
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/java/asiantech/internship/summer/LoginActivity.java]: `mPassword ` -> `password`

File: app/src/at-hungtran/java/asiantech/internship/summer/LoginActivity.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        mPattern = Pattern.compile(PASSWORD_PATTERN);
+        mMatcher = mPattern.matcher(mPassword);
+
+        return mMatcher.matches();
+
+    }
+
+    private boolean isValidEmail(String mEmail) {
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/java/asiantech/internship/summer/LoginActivity.java]: `mEmail ` -> `email`

File: app/src/at-hungtran/res/layout/activity_login.xml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginLeft="30dp"
+            android:layout_weight="7"
+            android:text="@string/signupActivity"
+            android:textColor="@color/colorWhite"
+            android:textSize="20sp" />
+
+        <Button
+            android:id="@+id/btn"
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/res/layout/activity_login.xml]: change id more clearly

File: app/src/at-hungtran/java/asiantech/internship/summer/eventlistener/LoginActivity.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        mEdtUser.addTextChangedListener(mTextWatcher);
+        mEdtPwd.addTextChangedListener(mTextWatcher);
+        mEdtEmail.addTextChangedListener(mTextWatcher);
+
+        // run once to disable if empty
+        checkUsernamePasswordandEmail();
+    }
+
+    private void checkUsernamePasswordandEmail() {
+        Button btnLogin = findViewById(R.id.btnLogin);
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/java/asiantech/internship/summer/eventlistener/LoginActivity.java]: define this variable as a global

File: app/src/at-hungtran/java/asiantech/internship/summer/eventlistener/LoginActivity.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+        String username = mEdtUser.getText().toString();
+        String password = mEdtPwd.getText().toString();
+        String email = mEdtEmail.getText().toString();
+        mEdtPwd.setTransformationMethod(PasswordTransformationMethod.getInstance());
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/java/asiantech/internship/summer/eventlistener/LoginActivity.java]: move this line to `onCreate` method

File: app/src/at-hungtran/res/values/strings.xml [hunk 7]
```diff
@@ -1,3 +1,13 @@
 <resources>
     <string name="app_name">Intern18Summer</string>
+    <string name="todo">TODO</string>
+    <string name="allow_email_from_other">Allow_email_from_other</string>
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/res/values/strings.xml]: `allow_email_from_other ` -> `allowEmailFromOther `

File: app/src/at-hungtran/res/values/strings.xml [hunk 8]
```diff
@@ -1,3 +1,13 @@
...
+    <string name="email_subscriptions">Email Subscriptions</string>
```
Comment (at-vinhhuynh) [inline app/src/at-hungtran/res/values/strings.xml]: `email_subscriptions` -> `emailSubscriptions`

**Pull request merged**